irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -44,6 +45,17 @@ namespace {
     }
   }
 
+  std::string formatPostgresMessage(const char *message) {
+    std::string formatted_message(message);
+    boost::replace_all(formatted_message, "\n", " ");
+    return formatted_message;
+  }
+
+  void processPqNotice(void *arg, const char *message) {
+    auto *log = reinterpret_cast<logger::Logger *>(arg);
```

Comment:
Please check if dynamic_cast can be employed here instead of reinterpret_cast.

Reply:
no, it cannot be employed with `void*`